### PR TITLE
Improvements to Miner & Validator

### PR DESCRIPTION
**Pull request by ibraheem-opentensor**
- Passes config instead of explicitly passing port in miner
- Makes use of subtensor in doing calls to the chain instead of direct SubstrateInterface
- This also fixes a bug where `subtensor.network` is passed as `local` but the nodes were querying finney due to default
- Fixes bug in validator with `moving_avg` in-case a new neuron joins the network

### CODE REVIEW

File: validator.py [hunk 1]
```diff
@@ -45,7 +50,7 @@ def get_config(self):
                 config.wallet.name,
                 config.wallet.hotkey_str,
                 config.netuid,
-                'validator',
+                "validator",
```
Comment (roman-opentensor) [inline validator.py:53]: could we move in into the string instead of providing as a argument?

File: validator.py [hunk 2]
```diff
@@ -80,29 +87,22 @@ def setup_bittensor_objects(self):
 
         # Connect the validator to the network.
         if self.wallet.hotkey.ss58_address not in self.metagraph.hotkeys:
-            bt.logging.error(f"\nYour validator: {self.wallet} is not registered to chain connection: {self.subtensor} \nRun 'btcli register' and try again.")
+            bt.logging.error(
+                f"\nYour validator: {self.wallet} is not registered to chain connection: {self.subtensor} \nRun 'btcli register' and try again."
```
Comment (roman-opentensor) [inline validator.py:91]: do we really need `\n` in the beginning of the message?

**Review by roman-opentensor — APPROVED** [review]
as a suggestion: it would be nice to use [blue][/blue] (or another depends by the case) tag for some values in the logging messages to be more consistent with btcli and sdk

**Comment by ibraheem-opentensor** [comment]
> as a suggestion: it would be nice to use [blue][/blue] (or another depends by the case) tag for some values in the logging messages to be more consistent with btcli and sdk

Agreed. Added this in set_weights so it can be distinct from others

**Review by roman-opentensor — APPROVED** [review]

**Pull request merged**